### PR TITLE
Add coach indicator tokens

### PR DESCRIPTION
**Pull request by lynnhao**
## Description

- Added coach indicator tokens, including both color and non-color

## Motivation and context

- Coach indicator (previously on contributions) is being formalized as a Spectrum 2 component. These net-new tokens define the design data needed for implementation. 

## Related issue

[SDS-13422](https://jira.corp.adobe.com/browse/SDS-13422)

## Types of changes

<li> [ ] Patch (bug fixes, typos, mistakes; non-breaking change which fixes an issue) </li>
<li> [x] Minor (add a new token, changing a value, deprecating a token; non-breaking change which adds functionality) </li>
<li> [ ] Major (deleting a token, changing token value type, renaming a token by deprecating the old one; fix or feature that would cause existing functionality to change) </li>

## Checklist:

<li> [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html). </li>
<li> [x] I updated the token in all applicable sets. This applies if updating, adding, or deleting a token that has data across different sets (for example, if the value differs across color themes.) </li>

### CODE REVIEW

**Comment by mrcjhicks** [comment]
**[TESTING spectrum2](https://github.com/adobe/spectrum-tokens-studio-data/actions/runs/9620082601)**

READING spectrum-tokens: `/spectrum-tokens/packages/tokens/src/`
READING   tokens studio: `/spectrum-tokens-studio-data/src/tokens-studio/spectrum2-colors/`
READING   tokens studio: `/spectrum-tokens-studio-data/src/tokens-studio/spectrum2-non-colors/`

<details><summary>STUDIO SYNC SUMMARY</summary><p>

```
UPDATED METADATA - SPECTRUM UUID: spectrum2/component.Component.coach-indicator
UPDATED METADATA - SPECTRUM UUID: spectrum2/layout.component.coach-indicator-collapsed-ring-rounding-increment
UPDATED METADATA - SPECTRUM UUID: spectrum2/layout.component.coach-indicator-expanded-ring-rounding-increment
UPDATED METADATA - SPECTRUM UUID: spectrum2/layout.component.coach-indicator-collapsed-ring-thickness
UPDATED METADATA - SPECTRUM UUID: spectrum2/layout.component.coach-indicator-expanded-ring-thickness
UPDATED METADATA - SPECTRUM UUID: spectrum2/layout.component.coach-indicator-collapsed-gap
UPDATED METADATA - SPECTRUM UUID: spectrum2/layout.component.coach-indicator-expanded-gap
UPDATED METADATA - SPECTRUM UUID: spectrum2/opacity.component.coach-indicator-opacity
```
</p></details>

<details><summary>SPECTRUM SYNC SUMMARY</summary><p>

```
NEW TOKENS: (12) contextual-help-body-font-size/desktop, contextual-help-title-font-size/desktop, contextual-help-body-font-size/mobile, contextual-help-title-font-size/mobile, coach-indicator-color, coach-indicator-collapsed-ring-rounding-increment, coach-indicator-expanded-ring-rounding-increment, coach-indicator-collapsed-ring-thickness, coach-indicator-expanded-ring-thickness, coach-indicator-collapsed-gap, coach-indicator-expanded-gap, coach-indicator-opacity
MODIFIED TOKENS: (2) contextual-help-body-size, contextual-help-title-size
DEPRECATED TOKENS: (6) contextual-help-title-size/desktop, contextual-help-title-size/mobile, contextual-help-body-size/desktop, contextual-help-body-size/mobile, contextual-help-body-size, contextual-help-title-size
CHANGED VALUES: (2) contextual-help-body-size/desktop, contextual-help-title-size/desktop
```
</p></details>

TEST COMPLETE

**Comment by lynnhao** [comment]
@GarthDB @karstens @mrcjhicks  – When you do your review, could you make sure to help check the "type" for the new tokens? I updated them based on other similar tokens, but some are new here.

For example, this update includes `increment` tokens – I'm assuming this is a `number` (similar to `multiplier` tokens), but would like to make sure this is correct.

Thanks!

**Review by nabuhasan — APPROVED** [review]
Overall no concerns about the new additions - just a question about the changed content in the $themes.json, why are those showing up?

File: src/tokens-studio/spectrum2-colors/$themes.json [hunk 1]
```diff
@@ -1335,8 +1338,7 @@
       "Palette.transparent-white.700": "fbac60c7cd862f46eba4908ae9cc5c7346156320",
       "Palette.transparent-white.800": "ea5e834dbe202dc1dbbce86ba074ab50155812c1",
       "Palette.transparent-white.900": "cc73dfce811bf9bab293a95d9342fd6a9135c9ab",
-      "Palette.transparent-white.1000": "0ae4fb367bd6d70103b8df1fd1fff8a237a94b30",
-      "Icon.primary.disabled.default": "99b973292918e2c767311e3bca5831040c7f9ae9"
+      "Palette.transparent-white.1000": "0ae4fb367bd6d70103b8df1fd1fff8a237a94b30"
     },
```
Comment (karstens) [inline src/tokens-studio/spectrum2-colors/$themes.json:1342]: @nabuhasan not sure, why the changes above appear here. Again, we are not in control about how FTS is syncing tokens and variables, but these changes are all okay. It is just the order of the entries, the "token name to variable id" mapping is correct. But thanks for the extra eyes. This data is really "sensitive" and we should be extra careful with these entries. 👍

File: src/tokens-studio/spectrum2-non-colors/spectrum2/layout.component/desktop.json [hunk 2]
```diff
@@ -1059,6 +1059,26 @@
       }
     }
   },
+  "coach-indicator-collapsed-ring-rounding-increment": {
+    "value": "6px",
+    "type": "number",
```
Comment (karstens) [inline src/tokens-studio/spectrum2-non-colors/spectrum2/layout.component/desktop.json:1064]: @lynnhao This should be "type": "spacing" or "type": "sizing". These tokens have a pixel value and therefor are _not_ Type: `number`.

File: src/tokens-studio/spectrum2-non-colors/spectrum2/layout.component/desktop.json [hunk 3]
```diff
@@ -1059,6 +1059,26 @@
...
+    "$extensions": {
+      "spectrum-tokens": {
+        "uuid": "",
+        "name": "coach-indicator-collapsed-ring-rounding-increment"
+      }
+    }
+  },
+  "coach-indicator-expanded-ring-rounding-increment": {
+    "value": "14px",
+    "type": "number",
```
Comment (karstens) [inline src/tokens-studio/spectrum2-non-colors/spectrum2/layout.component/desktop.json:1074]: @lynnhao This should be "type": "spacing" or "type": "sizing". These tokens have a pixel value and therefor are _not_ Type: `number`.

File: src/tokens-studio/spectrum2-non-colors/spectrum2/layout.component/mobile.json [hunk 4]
```diff
@@ -1059,6 +1079,26 @@
       }
     }
   },
+  "coach-indicator-collapsed-ring-rounding-increment": {
+    "value": "6px",
+    "type": "number",
```
Comment (karstens) [inline src/tokens-studio/spectrum2-non-colors/spectrum2/layout.component/mobile.json:1084]: @lynnhao This should be "type": "spacing" or "type": "sizing". These tokens have a pixel value and therefor are _not_ Type: `number`.

File: src/tokens-studio/spectrum2-non-colors/spectrum2/layout.component/mobile.json [hunk 5]
```diff
@@ -1059,6 +1079,26 @@
...
+    "$extensions": {
+      "spectrum-tokens": {
+        "uuid": "",
+        "name": "coach-indicator-collapsed-ring-rounding-increment"
+      }
+    }
+  },
+  "coach-indicator-expanded-ring-rounding-increment": {
+    "value": "14px",
+    "type": "number",
```
Comment (karstens) [inline src/tokens-studio/spectrum2-non-colors/spectrum2/layout.component/mobile.json:1094]: @lynnhao This should be "type": "spacing" or "type": "sizing". These tokens have a pixel value and therefor are _not_ Type: `number`.

**Review by karstens — CHANGES_REQUESTED** [review]
@lynnhao I left comments on the tokens that need a "type" change. We haven't really decided, which tokens we want to categorise as "spacing" and which as "sizing", but the token should not be a "number", if it has a pixel "px" in the value.

File: src/tokens-studio/spectrum2-colors/$themes.json (hunk 1 above)
Comment (nabuhasan) [reply src/tokens-studio/spectrum2-colors/$themes.json:1342]: Thanks for that clarification @karstens!

**Comment by karstens** [comment]
@lynnhao Just FYI: I resolved the merge conflicts in this PR, related the recent changes in the :main branch. ✅

**Comment by mrcjhicks** [comment]
**[TESTING spectrum2](https://github.com/adobe/spectrum-tokens-studio-data/actions/runs/9666136768)**

READING spectrum-tokens: `/spectrum-tokens/packages/tokens/src/`
READING   tokens studio: `/spectrum-tokens-studio-data/src/tokens-studio/spectrum2-colors/`
READING   tokens studio: `/spectrum-tokens-studio-data/src/tokens-studio/spectrum2-non-colors/`

  > WARNING - STUDIO INCLUDES DEPRECATED TOKEN: overlay-opacity/dark [31d5b502-6266-4309-8f8a-3892e6e158da]
  > WARNING - STUDIO INCLUDES DEPRECATED TOKEN: overlay-opacity/light [26b9803c-577f-4a29-badd-dfc459e8b73e]


<details><summary>STUDIO SYNC SUMMARY</summary><p>

```
UPDATED METADATA - SPECTRUM UUID: spectrum2/component.Component.coach-indicator
UPDATED METADATA - SPECTRUM UUID: spectrum2/layout.component.coach-indicator-collapsed-ring-rounding-increment
UPDATED METADATA - SPECTRUM UUID: spectrum2/layout.component.coach-indicator-expanded-ring-rounding-increment
UPDATED METADATA - SPECTRUM UUID: spectrum2/layout.component.coach-indicator-collapsed-ring-thickness
UPDATED METADATA - SPECTRUM UUID: spectrum2/layout.component.coach-indicator-expanded-ring-thickness
UPDATED METADATA - SPECTRUM UUID: spectrum2/layout.component.coach-indicator-collapsed-gap
UPDATED METADATA - SPECTRUM UUID: spectrum2/layout.component.coach-indicator-expanded-gap
UPDATED METADATA - SPECTRUM UUID: spectrum2/opacity.component.coach-indicator-opacity
```
</p></details>

<details><summary>SPECTRUM SYNC SUMMARY</summary><p>

```
NEW TOKENS: (8) coach-indicator-color, coach-indicator-collapsed-ring-rounding-increment, coach-indicator-expanded-ring-rounding-increment, coach-indicator-collapsed-ring-thickness, coach-indicator-expanded-ring-thickness, coach-indicator-collapsed-gap, coach-indicator-expanded-gap, coach-indicator-opacity
```
</p></details>

TEST COMPLETE

File: src/tokens-studio/spectrum2-non-colors/spectrum2/layout.component/desktop.json (hunk 2 above)
Comment (lynnhao) [reply src/tokens-studio/spectrum2-non-colors/spectrum2/layout.component/desktop.json]: This has been fixed and updated to `borderRadius` 👍

File: src/tokens-studio/spectrum2-non-colors/spectrum2/layout.component/mobile.json (hunk 4 above)
Comment (lynnhao) [reply src/tokens-studio/spectrum2-non-colors/spectrum2/layout.component/mobile.json]: This has been fixed and updated to `borderRadius` 👍

File: src/tokens-studio/spectrum2-non-colors/spectrum2/layout.component/mobile.json (hunk 5 above)
Comment (lynnhao) [reply src/tokens-studio/spectrum2-non-colors/spectrum2/layout.component/mobile.json]: This has been fixed and updated to `borderRadius` 👍

**Comment by lynnhao** [comment]
Thank you @karstens for the detailed review. I've updated the token types of the "increment" tokens to `borderRadius` as discussed on Slack. These should be ready now.

**Comment by GarthDB** [comment]
> @lynnhao I left comments on the tokens that need a "type" change. We haven't really decided, which tokens we want to categorise as "spacing" and which as "sizing", but the token should not be a "number", if it has a pixel "px" in the value.

@karstens We could put together tests to run to check for this if that helps in the future.
